### PR TITLE
ENH: Add support for reading a requested subregion

### PR DESCRIPTION
**Pull request by tbirdso**
Adds support for reading a requested image subregion from an OME-Zarr NGFF image with `tensorstore`.

Includes testing and baseline comparison.

### CODE REVIEW

File: src/itkOMEZarrNGFFImageIO.cxx [hunk 1]
```diff
@@ -433,41 +483,46 @@ OMEZarrNGFFImageIO::ReadImageInformation()
...
   }
 
 void
 OMEZarrNGFFImageIO::Read(void * buffer)
 {
+  // Use a proxy measure (voxel count) to determine whether we are reading
+  // the entire image or an image subregion.
   // This comparison needs to be done carefully, we can compare 3D and 6D regions
-  if (this->GetLargestRegion().GetNumberOfPixels() != m_IORegion.GetNumberOfPixels())
+  if (this->GetLargestRegion().GetNumberOfPixels() == m_IORegion.GetNumberOfPixels())
```
Comment (dzenanz) [inline src/itkOMEZarrNGFFImageIO.cxx:498]: Will this not cause an exception if entire image is read at once?

Comment (tbirdso) [reply src/itkOMEZarrNGFFImageIO.cxx:498]: No, this block is updated to simply verify that the size of the store matches the requested IO region. Existing CTests reading the entire image pass on my local system, will watch CI to verify.

**Review by thewtex — APPROVED** [review]
:1st_place_medal:

**Comment by tbirdso** [comment]
Moving forward with merge in support of subsequent changes. @dzenanz Please let me know if you have additional feedback following merge, or add discussion in https://github.com/InsightSoftwareConsortium/ITKIOOMEZarrNGFF/issues/23 .

**Pull request merged**